docs/aws/audit/redshiftmonitoring/rules/default_master_username.mdx

```diff
@@ -24,6 +24,249 @@ PCIDSS
+   Use the AWS SDK for Python (Boto3) to automate the creation of a Redshift cluster with a custom master username.
+
+   ```python
+   import boto3
+
+   redshift = boto3.client('redshift')
+
+   response = redshift.create_cluster(
+       ClusterIdentifier='my-cluster',
+       NodeType='dc2.large',
+       MasterUsername='customadmin',
+       MasterUserPassword='MySecurePassword123',
+       ClusterType='single-node'
+   )
+
+   print(response)
+   ```
+
+   Run the Python script to create the cluster:
+
+   ```sh
+   python create_redshift_cluster.py
+   ```
+
+By following these steps, you can ensure that your Redshift clusters do not use the default master username, thereby enhancing the security of your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the use of the default master username in Amazon Redshift clusters using Python scripts, you can use the AWS SDK for Python (Boto3). Below are the steps to achieve this:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Python Script**:
+   Write a Python script that will create a Redshift cluster with a custom master username instead of the default one.
+
+3. **Define the Custom Master Username**:
+   Specify a custom master username in your script to ensure it does not use the default username.
+
+4. **Create the Redshift Cluster**:
+   Use the Boto3 `create_cluster` method to create the Redshift cluster with the custom master username.
+
+Here is a sample Python script to prevent using the default master username:
+
+```python
+import boto3
+from botocore.exceptions import ClientError
+
+# Initialize a session using Amazon Redshift
+redshift = boto3.client('redshift')
+
+# Define the cluster parameters
+cluster_identifier = 'my-redshift-cluster'
+node_type = 'dc2.large'
+master_username = 'customadmin'  # Custom master username
+master_user_password = 'YourSecurePassword123'
+db_name = 'mydatabase'
+cluster_type = 'single-node'
+
+try:
+    # Create the Redshift cluster
+    response = redshift.create_cluster(
+        ClusterIdentifier=cluster_identifier,
+        NodeType=node_type,
+        MasterUsername=master_username,
+        MasterUserPassword=master_user_password,
+        DBName=db_name,
+        ClusterType=cluster_type
+    )
+    print(f"Cluster {cluster_identifier} created successfully.")
+except ClientError as e:
+    print(f"Error creating cluster: {e}")
+```
+
+### Key Points:
+1. **Install Boto3**: Ensure Boto3 is installed in your Python environment.
+2. **Define Custom Master Username**: Specify a custom master username in the script.
+3. **Create Redshift Cluster**: Use the `create_cluster` method with the custom master username.
+4. **Error Handling**: Implement error handling to catch and display any issues during cluster creation.
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>